MobiFlight/BoardDefinitions.cs

```diff
@@ -84,8 +86,27 @@ public static Board GetBoardByMobiFlightType(String mobiflightType)
         /// </summary>
         public static void LoadDefinitions()
         {
-            boards = JsonBackedObject.LoadDefinitions<Board>(Directory.GetFiles("Boards", "*.board.json"), "Boards/mfboard.schema.json",
-                onSuccess: board => Log.Instance.log($"Loaded board definition for {board.Info.MobiFlightType} ({board.Info.FriendlyName})", LogSeverity.Info),
+            var files = new List<String>(Directory.GetFiles("Boards", "*.board.json"));
+            files.AddRange(Directory.GetFiles("Community/", "*.board.json", SearchOption.AllDirectories));
+            boards = JsonBackedObject.LoadDefinitions<Board>(files.ToArray(), "Boards/mfboard.schema.json",
+                onSuccess: (board, definitionFile) => {
+                    Log.Instance.log($"Loaded board definition for {board.Info.MobiFlightType} ({board.Info.FriendlyName})", LogSeverity.Info);
+
+                    var boardPath = Path.GetDirectoryName(definitionFile);
+                    board.BasePath = Path.GetDirectoryName(boardPath);
+
+                    var logoPath = $@"{board.BasePath}\logo.png";
+                    if (File.Exists(logoPath) && board.Info.Community != null)
+                    {
+                        board.Info.Community.Logo = Image.FromFile(logoPath);
+                    }
+
+                    var boardLogoPath = $@"{boardPath}\board-logo.png";
+                    if (File.Exists(boardLogoPath))
+                    {
+                        board.BoardImage = Image.FromFile(boardLogoPath);
+                    }
```

Comment:
Thinking about this on my route, I'm not wild about this getting set here instead of being done as getters in the actual objects (like you're already doing for `PartnerLevel`). This stuff is really specific to the Board, not the class that handles loading files.

I took a pass at moving it to getters and doing `BoardImage` is easy since it's right on the main object and `BasePath` is accessible. I got stuck on doing it for the community logo though since that's buried two classes deep from the board and doesn't have access to `BasePath`. Any ideas how that could be done?

Reply:
I put it on purpose there, because i didn't want to make the board class so smart... so far it was more like a simple Data Object filled with the info from the JSON.

Since the infos don't come from the JSON, i thought it would be better to have the Definition loader handle this extra logic and provide it from the outside.

I still like the idea.

Reply:
You already have logic in the Board class that is beyond just JSON backed data: `PartnerLevel`. The code is inconsistent now, where some of the calculated properties happen on the fly while others are set by an outside class.